src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -207,105 +227,103 @@ private void appendOneNativeArgument(ExecutionContext context, object obj, char
-                                            if (! first)
-                                            {
-                                                _arguments.Append(" ");
-                                            }
-                                            else
-                                            {
-                                                if (! (pbo is System.IO.FileSystemInfo))
-                                                {
-                                                    // If the object is not a filesystem object, then just append
-                                                    // the pattern unchanged
-                                                    _arguments.Append(arg);
-                                                    break;
-                                                }
-                                                first = false;
-                                            }
-                                            var expandedPath = (pbo as System.IO.FileSystemInfo).FullName;
-                                            if (normalizePath)
-                                            {
-                                                expandedPath = Context.SessionState.Path.NormalizeRelativePath(expandedPath, cwdinfo.ProviderPath);
-                                            }
-                                            // If the path contains spaces, then add quotes around it.
-                                            if (NeedQuotes(expandedPath))
-                                            {
-                                                _arguments.Append("\"");
-                                                _arguments.Append(expandedPath);
-                                                _arguments.Append("\"");
-                                            }
-                                            else
-                                            {
-                                                _arguments.Append(expandedPath);
-                                            }
-                                        }
-                                    }
-                                    else
-                                    {
-                                        _arguments.Append(arg);
-                                    }
-                                }
-                                else
-                                {
-                                    _arguments.Append(arg);
-                                }
+                    // Expand paths, but only from the file system.
+                    if (paths?.Count > 0 && paths.All(p => p.BaseObject is FileSystemInfo))
+                    {
+                        var sep = "";
+                        foreach (var path in paths)
+                        {
+                            _arguments.Append(sep);
+                            sep = " ";
```

Comment:
Is `sep = " "` really needed here? It seems we just need `var sep = " "` right before the `foreach`

Reply:
Well, it's more tidy - it avoids putting 2 spaces between the previous token - either the command name or an argument.

Reply:
Thanks for clarifying #Closed.